assets/src/edit-story/elements/video/edit.js

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import styled from 'styled-components';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { elementFillContent, getMediaProps } from '../shared';
+import { useStory } from '../../app';
+import StoryPropTypes from '../../types';
+import EditPanMovable from '../shared/editPanMovable';
+import { videoWithScale } from './util';
+import ScalePanel from './scalePanel';
+
+const Element = styled.div`
+	${ elementFillContent }
+`;
+
+const CropBox = styled.div`
+	width: 100%;
+	height: 100%;
+	position: relative;
+	overflow: hidden;
+
+	&::after {
+		content: '';
+		display: block;
+		position: absolute;
+		left: 0;
+		top: 0;
+		width: 100%;
+		height: 100%;
+		border: 1px solid ${ ( { theme } ) => theme.colors.mg.v1 }70;
+		pointer-events: none;
+	}
+`;
+
+const FadedVideo = styled.video`
+	position: absolute;
+	opacity: 0.4;
+	pointer-events: none;
+	${ videoWithScale }
+	max-width: initial;
+	max-height: initial;
+`;
+
+const CropVideo = styled.video`
+	position: absolute;
+	${ videoWithScale }
+	max-width: initial;
+	max-height: initial;
+`;
+
+function VideoEdit( {
+	element: { id, src, origRatio, scale, focalX, focalY, mimeType },
+	box: { x, y, width, height, rotationAngle },
+} ) {
+	const [ fullVideo, setFullVideo ] = useState( null );
+	const [ croppedVideo, setCroppedVideo ] = useState( null );
+
+	const { actions: { updateElementById } } = useStory();
+	const setProperties = useCallback(
+		( properties ) => updateElementById( { elementId: id, properties } ),
+		[ id, updateElementById ] );
+
+	const videoProps = getMediaProps( width, height, scale, focalX, focalY, origRatio );
+
+	return (
+		<Element>
```

Comment:
Feels like we could do a `MediaElement` templates that could be reused by image, video, etc. But totally ok to do this and come back to this in later PRs if we find it necessary.

Reply:
Yes, seems so. Left this purposefully out for now until the video element's UX is better defined, there's a todo note in the description of this PR to reuse more code between the media elements as well.

Reply:
Makes sense.